website/promotion/forms.py

```diff
@@ -0,0 +1,33 @@
+from inspect import getmodule
+from django import forms
+from promotion.models import PromotionRequest
+from django.utils import timezone
+
+class PromotionRequestForm(forms.ModelForm):
+
+    class Meta:
+        model = PromotionRequest
+        fields = [
+            "event",
+            "publish_date",
+            "channel",
+            "assigned_to",
+            "status",
+            "drive_folder",
+            "remarks",
+        ]
+    
+    def clean_publish_date(self):
+        publish_date = self.cleaned_data.get("publish_date")
+        create_time_minimum = publish_date - timezone.timedelta(weeks=1)
```

Comment:
Let's make this timedelta a setting, because it feels like something that might change in the future